testsuite/testsuite.py

```diff
@@ -110,40 +110,11 @@ def tear_down(self):
 
         super(ALSTestsuite, self).tear_down()
 
-    def get_test_list(self, sublist):
-        results = []
-
-        blacklist = {os.path.abspath(os.path.join(self.test_dir, item))
-                     for item in self.TEST_BLACKLIST}
-        ada_lsp_dir = os.path.abspath(os.path.join(self.test_dir, 'ada_lsp'))
-
-        for dirpath, dirnames, filenames in os.walk(self.test_dir):
-            dirpath = os.path.abspath(dirpath)
-
-            # Ignore paths in the blacklist
-            if any(dirpath.startswith(item) for item in blacklist):
-                continue
-
-            # Warn about ada_lsp sub-directories that have no test.yaml file
-            if 'test.yaml' in filenames:
-                results.append(os.path.relpath(
-                    os.path.join(dirpath, 'test.yaml'),
-                    self.test_dir))
-
-            elif os.path.dirname(dirpath) == ada_lsp_dir:
-                print ("No test.yaml in {}".format(dirpath))
-                logging.warn('No test.yaml in %s', dirpath)
-
-        # If requested, keep only testcases that match one sublist item
-        if sublist:
-            logging.info('Filtering tests: %s', sublist)
-            results = [t for t in results
-                       if any(s in t for s in sublist)]
-
-        logging.info('Found %s tests', len(results))
-        logging.debug('tests:%s', '\n'.join('  ' + r for r in results))
-        return results
+    @property
+    def test_driver_map(self):
+        return {'ada_lsp': JsonTestDriver,
+                'codecs': CodecsTestDriver}
```

Comment:
You can now remove the `DRIVERS`, `TEST_SUBDIR` and `TEST_BLACKLIST` class attributes.